bitnami/phabricator/values.yaml

```diff
@@ -285,65 +517,135 @@ metrics:
-  type: ""
-  ## Node label key to match
-  ## E.g.
-  ## key: "kubernetes.io/e2e-az-name"
+externalDatabase:
+  ## Use existing secret (ignores previous password)
+  ## NOTE: must contain key `mariadb-root-password`
+  ## NOTE: When it's set, the `externalDatabase.rootPassword` parameter is ignored
   ##
-  key: ""
-  ## Node label values to match
-  ## E.g.
-  ## values:
-  ##   - e2e-az1
-  ##   - e2e-az2
+  # existingSecret:
+
+  ## Database host
   ##
-  values: []
+  host: localhost
 
-## Affinity for pod assignment. Evaluated as a template.
-## Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
-## Note: podAffinityPreset, podAntiAffinityPreset, and nodeAffinityPreset will be ignored when it's set
-##
-affinity: {}
+  ## Database port number
+  ##
+  port: 3306
 
-## Node labels for pod assignment. Evaluated as a template.
-## ref: https://kubernetes.io/docs/user-guide/node-selection/
-##
-nodeSelector: {}
+  ## root username of MariaDB
+  ##
+  rootUser: root
+
+  ## Database password
+  ##
+  rootPassword: ""
 
-## Tolerations for pod assignment. Evaluated as a template.
-## ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
 ##
-tolerations: []
+## MariaDB chart configuration
+##
+## https://github.com/bitnami/charts/blob/master/bitnami/mariadb/values.yaml
+##
+mariadb:
+  ## Whether to deploy a MariaDB server to satisfy the applications database requirements. To use an external database set this to false and configure the externalDatabase parameters
+  ##
+  enabled: true
+
+  ## MariaDB architecture. Allowed values: standalone or replication
+  ##
+  architecture: standalone
+
+  ## MariaDB Authentication parameters
+  ##
+  auth:
+    ## MariaDB root password
+    ## ref: https://github.com/bitnami/bitnami-docker-mariadb#setting-the-root-password-on-first-run
+    ##
+    rootPassword: ""
+
+  primary:
+    ## Improvements for better database performance on Phabricator
+    ##
+    configuration: |-
+      [mysqld]
+      skip-name-resolve
+      explicit_defaults_for_timestamp
+      basedir=/opt/bitnami/mariadb
+      plugin_dir=/opt/bitnami/mariadb/plugin
+      port=3306
+      socket=/opt/bitnami/mariadb/tmp/mysql.sock
+      tmpdir=/opt/bitnami/mariadb/tmp
+      max_allowed_packet = 32M
+      innodb_buffer_pool_size = 256M
+      bind-address=0.0.0.0
+      pid-file=/opt/bitnami/mariadb/tmp/mysqld.pid
+      log-error=/opt/bitnami/mariadb/logs/mysqld.log
+      character-set-server=UTF8
+      collation-server=utf8_general_ci
+      sql_mode = STRICT_ALL_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION
+
+      [client]
+      port=3306
+      socket=/opt/bitnami/mariadb/tmp/mysql.sock
+      default-character-set=UTF8
+      plugin_dir=/opt/bitnami/mariadb/plugin
+
+      [manager]
+      port=3306
+      socket=/opt/bitnami/mariadb/tmp/mysql.sock
+      pid-file=/opt/bitnami/mariadb/tmp/mysqld.pid
```

Comment:
It would be great if we could rely on envvars/extraFlags for Phabricator specifics, to avoid having to mount the entire file.

However, I understand we already mount the my.cnf by default so maybe it makes sense anyways to do it this way.

Reply:
We already mount a config file in MariaDB by default. This just replaces the original one with one that meets PH requirements.
Also, this is specific configuration of MySQL/MariaDB

Reply:
Nvm then, this should be fine